core/src/main/scala/kafka/log/Log.scala

```diff
@@ -753,6 +753,7 @@ class Log(@volatile var dir: File,
     debug("Closing log")
     lock synchronized {
       checkIfMemoryMappedBufferClosed()
+      scheduler.removeTask(task)
```

Comment:
Did we consider `task.cancel` instead of this approach:?

Reply:
We would also have to call `setRemoveOnCancelPolicy(true)` on the executor, but that seems like a good idea anyway.

Reply:
I didn't see this on the first time around. I can use this method instead.